binaryaudit/poky.py

```diff
@@ -120,3 +126,122 @@ def recipe_abicheck(recipe_binaudit_path, buildhistory_baseline_dir, bulidhistor
         res_details = out
 
     return item_name, base_version, new_version, exec_time, result, res_details, ret_acc
+
+
+def poky_binaryaudit(all_suppressions):
+    db_conn = None
+    if 'y' == args.enable_telemetry:
+        db_conn = connect_database()
+
+    if args.compare_buildhistory:
+        compare_buildhistory(all_suppressions, db_conn)
+
+    elif args.insert_baseline:
+        insert_baseline(db_conn)
+
+
+def connect_database():
+    try:
+        cli.validate_telemetry_args(args)
+    except argparse.ArgumentError as e:
+        util.fatal(str(e))
+        sys.exit(3)
+
+    from binaryaudit.db import wrapper as db_wrapper
+    try:
+        db_conn = db_wrapper(args.db_config, util.logger)
+        db_conn.initialize_db()
+    except Exception as e:
+        util.error(str(e))
+
+    return db_conn
+
+
+def compare_buildhistory(all_suppressions, db_conn):
+    d1 = args.buildhistory_baseline
+    d2 = args.buildhistory_current
+
+    out_dir = args.output_dir
+    # XXX err out if exists or remove existing?
+    if out_dir and not os.path.isdir(out_dir):
+        os.makedirs(out_dir)
+
+    if not 'y' == args.enable_telemetry and not os.path.isdir(d1):
+        util.warn("Directory '{}' doesn't exist.".format(d1))
+        sys.exit(1)
```

Comment:
Could these exits be please converted to `return` and the actual exit handled in `poky_binaryaudit`? This would be a small tweak that should not over complicate the patch a lot.

Thanks